modules/yardstick/src/main/java/org/apache/ignite/yardstick/cache/IgniteTxSuspendResumeBenchmark.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.yardstick.cache;
+
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteTransactions;
+import org.apache.ignite.transactions.Transaction;
+import org.apache.ignite.transactions.TransactionState;
+import org.apache.ignite.yardstick.cache.model.SampleValue;
+import org.yardstickframework.BenchmarkConfiguration;
+
+/**
+ * Ignite benchmark that performs suspend\resume operations.
+ * Concurrent threads must operate different keys, so there will be no waiting on key locks.
+ * So, all keys are separated on different groups, defined by first key and last key(first key plus keys number per thread).
+ * Groups have no intersections.
+ */
+public class IgniteTxSuspendResumeBenchmark extends IgniteCacheAbstractBenchmark<Integer, Object> {
+    /** Transactions. */
+    private IgniteTransactions txs;
+
+    /** ExecutorService. */
+    private ExecutorService ex;
+
+    /**
+     * First keyCounter that can be used by transaction in certain thread.
+     * The last one can be calculated based on keys per thread
+     */
+    private ThreadLocal<Integer> firstKey = new ThreadLocal<>();
```

Comment:
Maybe better use `ThreadLocal.withInitial()`?

Reply:
Or just override initialValue() in ThreadLocal. Tnx

Reply:
fixed